multiversx_sdk_core/proto/transaction_serializer.py

```diff
@@ -0,0 +1,93 @@
+from typing import Protocol
+
+import multiversx_sdk_core.proto.transaction_pb2 as ProtoTransaction
+from multiversx_sdk_core.address import AddressConverter
+from multiversx_sdk_core.constants import TRANSACTION_OPTIONS_TX_GUARDED
+
+
+class ITransaction(Protocol):
+    sender: str
+    receiver: str
+    gas_limit: int
+    chainID: str
+    gas_price: int
+    sender_username: str
+    receiver_username: str
+    nonce: int
+    amount: int
+    data: bytes
+    version: int
+    signature: bytes
+    options: int
+    guardian: str
+    guardian_signature: bytes
+
+
+class ProtoSerializer:
+    def serialize_transaction(self, transaction: ITransaction) -> bytes:
```

Comment:
We can define a constructor for `ProtoSerializer` and possibly receive `AddressConverter` as an interface in the constructor - I am thinking about having a HRP different than `erd` on side-chains etc.

I see that `TransactionComputer` instantiates the `ProtoSerializer`. The computer itself can also receive the address converter as an interface (I think that way will be implemented in Go).

While end-users can do `TransactionComputer(address_converter=AddressConverter())`.

Reply:
Done!

Reply:
Done it this way! Thanks